metrics.go

```diff
@@ -35,101 +44,136 @@ var (
+		Buckets: speedBytesPerMsHistogram,
+	}, []string{"resourceType", "cache_status"})
+
+	fetchCacheCountSuccessTotalMetric = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Name: prometheus.BuildFQName("ipfs", "caboose", "fetch_cache_count_success_total"),
+		Help: "Records cache hits and cache hits for successful fetches from Saturn",
+	}, []string{"resourceType", "cache_status"})
+)
 
-	fetchDurationPerBlockPerPeerSuccessMetric = prometheus.NewHistogram(prometheus.HistogramOpts{
+// block metrics
+var (
+	fetchDurationPerBlockPerPeerSuccessMetric = prometheus.NewHistogramVec(prometheus.HistogramOpts{
 		Name:    prometheus.BuildFQName("ipfs", "caboose", "fetch_duration_block_peer_success"),
-		Help:    "Latency observed during successful caboose fetches from a single peer",
-		Buckets: durationPerBlockPerPeerHistogram,
-	})
+		Help:    "Latency observed during successful caboose fetches from a single peer in milliseconds",
+		Buckets: durationMsPerBlockHistogram,
+	}, []string{"cache_status"})
 
-	fetchDurationPerCarPerPeerSuccessMetric = prometheus.NewHistogram(prometheus.HistogramOpts{
-		Name:    prometheus.BuildFQName("ipfs", "caboose", "fetch_duration_car_peer_success"),
-		Help:    "Latency observed during successful caboose car fetches from a single peer",
-		Buckets: durationPerCarHistogram,
+	fetchDurationBlockSuccessMetric = prometheus.NewHistogram(prometheus.HistogramOpts{
+		Name:    prometheus.BuildFQName("ipfs", "caboose", "fetch_duration_block_success"),
+		Help:    "Latency observed during successful caboose fetches for a block across multiple peers and retries in milliseconds",
+		Buckets: durationMsPerBlockHistogram,
 	})
 
+	fetchTTFBPerBlockPerPeerSuccessMetric = prometheus.NewHistogramVec(prometheus.HistogramOpts{
+		Name:    prometheus.BuildFQName("ipfs", "caboose", "fetch_ttfb_block_peer_success"),
+		Help:    "TTFB observed during a successful caboose fetch from a single peer in milliseconds",
+		Buckets: durationMsPerBlockHistogram,
+	}, []string{"cache_status"})
+
+	// failures
 	fetchDurationPerBlockPerPeerFailureMetric = prometheus.NewHistogram(prometheus.HistogramOpts{
 		Name:    prometheus.BuildFQName("ipfs", "caboose", "fetch_duration_block_peer_failure"),
-		Help:    "Latency observed during failed caboose fetches from a single peer",
-		Buckets: durationPerBlockPerPeerHistogram,
+		Help:    "Latency observed during failed caboose fetches from a single peer in milliseconds",
+		Buckets: durationMsPerBlockHistogram,
 	})
 
-	fetchDurationPerCarPerPeerFailureMetric = prometheus.NewHistogram(prometheus.HistogramOpts{
-		Name:    prometheus.BuildFQName("ipfs", "caboose", "fetch_duration_car_peer_failure"),
-		Help:    "Latency observed during failed caboose car fetches from a single peer",
-		Buckets: durationPerCarHistogram,
+	fetchDurationBlockFailureMetric = prometheus.NewHistogram(prometheus.HistogramOpts{
+		Name:    prometheus.BuildFQName("ipfs", "caboose", "fetch_duration_block_failure"),
```

Comment:
renaming current metrics will break current dashboards - why is this changing?

Reply:
@willscott Not all the metrics have been renamed. Only some that were either incorrectly named or some that have taken the place of some repetitive ones.

I'll ensure the Caboose/Bifrost dashboard/metrics encompass the correct names. We're anyways revamping both the dashboards.